src/test/scala/com/mongodb/spark/rdd/partitioner/BsonValueOrderingSpec.scala

```diff
@@ -133,34 +129,39 @@ class BsonValueOrderingSpec extends FlatSpec with Matchers {
       new BsonRegularExpression("[0-9a-z]+"),
       new BsonRegularExpression("[a-z]+")
     )
-    val ordered = Random.shuffle(data).sorted
-    ordered should equal(data)
+    testAllPermutations(data)
   }
 
   it should "compare document types correctly" in {
     val data: Seq[BsonValue] = Seq(
       BsonDocument.parse("""{}"""),
       BsonDocument.parse("""{a: -1}"""),
-      BsonDocument.parse("""{a: 0}"""),
       BsonDocument.parse("""{a: 1}"""),
       BsonDocument.parse("""{a: 1, b: 1}"""),
-      BsonDocument.parse("""{a: 2}"""),
-      BsonDocument.parse("""{a: 9, b: 1}"""),
-      BsonDocument.parse("""{a: 10, b: 0}"""),
-      BsonDocument.parse("""{a: 100, b: 10}"""),
-      BsonDocument.parse("""{a: 1000, b: 10}"""),
-      BsonDocument.parse("""{a: 1000, b: 10, c: -1}"""),
-      BsonDocument.parse("""{a: "1", b: 0}""")
+      BsonDocument.parse("""{a: 1, b: 2}"""),
+      BsonDocument.parse("""{a: 1, b: 2, c: -1}"""),
+      BsonDocument.parse("""{a: "1", b: 0}"""),
+      BsonDocument.parse("""{a: [1, 2], b: 0}""")
     )
-    val ordered = Random.shuffle(data).sorted
-    ordered should equal(data)
+
+    testAllPermutations(data)
   }
 
   it should "have no defined order for undefined types" in {
     val data: Seq[BsonValue] = allBsonTypes.filter({ case kv => undefinedOrderingKeys.contains(kv._1) }).map(_._2)
-    val randomised = Random.shuffle(data)
-    val ordered = randomised.sorted
-    ordered should equal(randomised)
+
+    for (perm <- data.permutations) {
+      val randomised = Random.shuffle(perm)
+      val ordered = randomised.sorted
+      ordered should equal(randomised)
+    }
+  }
+
+  def testAllPermutations(data: Seq[BsonValue]): Unit = {
```

Comment:
We now ensure we test all order permutations as the older random shuffle approach could easily miss some ordering collisions.